packages/@aws-cdk/pipelines/ORIGINAL_API.md

```diff
@@ -2,8 +2,187 @@
+      'cdk synth',
+    ],
+
+    // etc...
+  }),
+})
+```
+
+Instead of specifying the pipeline source with the `sourceAction` property to
+the pipeline, specify it as the `input` property to the `ShellStep` instead.
+You can use any of the factory functions on `CodePipelineSource`.
+
+For example, for a GitHub source, the following:
+
+```ts
+sourceAction: new codepipeline_actions.GitHubSourceAction({
+  actionName: 'GitHub',
+  output: sourceArtifact,
+  // Replace these with your actual GitHub project name
+  owner: 'OWNER',
+  repo: 'REPO',
+  branch: 'main', // default: 'master'
+}),
+```
+
+Translates into:
+
+```ts
+input: CodePipelineSource.gitHub('OWNER/REPO', 'main', {
+  authentication: SecretValue.secretsManager('GITHUB_TOKEN_NAME'),
+}),
+```
+
+### Deployments
+
+Adding CDK Stages to deploy is done by calling `addStage()`, or
+potentially `addWave().addStage()`. All stages inside a wave are
+deployed in parallel, which was not a capability of the original API.
+
+ Old method | New method
+------------|------------------
+`addApplicationStage()` | ⟹ `addStage()`
+`addStage().addApplication()` | ⟹ `addStage()`. Adding multiple CDK Stages into a single Pipeline stage is not supported, add multiple Pipeline stages instead.
+
+### Approvals
+
+Approvals are added by adding `pre` and `post` options to `addStage()`, with
+steps to execute before and after the deployments, respectively. We recommend
+putting manual approvals in `pre` steps, and automated approvals in `post` steps.
+
+#### Manual approvals
+
+```ts
+const stage = pipeline.addApplicationStage(...);
+stage.addAction(new ManualApprovalAction({
+  actionName: 'ManualApproval',
+  runOrder: testingStage.nextSequentialRunOrder(),
+}));
+```
+
+Becomes:
+
+```ts
+pipeline.addStage(..., {
+  pre: [
+    new ManualApprovalStep('ManualApproval'),
+  ],
+});
+```
+
+#### Automated approvals
+
+```ts
+const stage = pipeline.addApplicationStage(...);
+stage.addActions(new ShellScriptAction({
+  actionName: 'MyValidation',
+  commands: ['curl -Ssf $VAR'],
+  useOutputs: {
+    VAR: pipeline.stackOutput(stage.cfnOutput),
+  },
+  // Optionally specify a BuildEnvironment
+  environment: { ... },
+}));
+```
+
+Becomes:
+
+```ts
+const stage = new MyStage(...);
+pipeline.addStage(stage, {
+  post: [
+    new CodeBuildStep('MyValidation', {
+      commands: ['curl -Ssf $VAR'],
+      envFromCfnOutput: {
+        VAR: stage.cfnOutput,
+      },
+      // Optionally specify a BuildEnvironment
+      buildEnvironment: { ... },
+    }),
+  ],
+});
+```
+
+#### Change set approvals
+
+There are two properties that were used to add actions to the pipeline
+in between the `CreateChangeSet` and `ExecuteChangeSet` actions:
+
+```ts
+    manualApprovals: true,
+    extraRunOrderSpace: 1,
+```
+
+These are not supported in the new API.
+
+### Custom CodePipeline Actions
+
+See the section [**Arbitrary CodePipeline actions** in the
+main `README`](README.md) for an example of how to inject arbitrary
```

Comment:
can we add an anchor link?

Reply:
I guess we can, but I don't like anchor links to markdowns: they're implementation-depdendent. OTOH, we're definitely on GitHub anyway...